app/llama_sensei/backend/qa/generate_answer.py

```diff
@@ -57,7 +57,10 @@ def __init__(self, course: str, model=MODEL):
         self.embedder = SentenceTransformer(EMBEDDING_LLM, trust_remote_code=True).to(
             DEVICE
         )
-        self.model = ChatGroq(model=model, temperature=0)
+        self.groq_api_key = groq_api_key
+        self.model = ChatGroq(
+            model=model, temperature=0, groq_api_key=self.groq_api_key
```

Comment:
@thaohp113 
API key can be accidentally disclosed if loading like your.
Should load it from `.env` and consider to use `python-dotenv` if you don't want to export it manually.

Reply:
solved by using os.env
`import os
os.environ["GROQ_API_KEY"] = "valid_key"
GROQ_API_KEY = os.environ.get("GROQ_API_KEY")`